text/0034-xcm-absolute-location-account-derivation.md

```diff
@@ -0,0 +1,199 @@
+			origin, instructions, max_weight, properties,
+		);
+		let mut actual_origin = *origin;
+		let skipped = Cell::new(0usize);
+		// NOTE: We do not check the validity of `UniversalOrigin` here, meaning that a malicious
+		// origin could place a `UniversalOrigin` in order to spoof some location which gets free
+		// execution. This technical could get it past the barrier condition, but the execution
+		// would instantly fail since the first instruction would cause an error with the
+		// invalid UniversalOrigin.
+		instructions.matcher().match_next_inst_while(
+			|_| skipped.get() < MaxPrefixes::get() as usize,
+			|inst| {
+				match inst {
+					UniversalOrigin(new_global) => {
+-						// Note the origin is *relative to local consensus*! So we need to escape
+-						// local consensus with the `parents` before diving in into the
+-						// `universal_location`.
+-						actual_origin = X1(*new_global).relative_to(&LocalUniversal::get());
++                        // Grab the GlobalConsensus junction of LocalUniversal.
++                        if let Ok(this_global) = LocalUniversal::get().global_consensus() {
++                            // Error if requested GlobalConsensus is not this location's GlobalConsensus.
++                            if *new_global != this_global.into() {
++                                return Err(ProcessMessageError::Unsupported);
++                            }
++
++                            // Build a location with 2 parents.
++                            actual_origin = MultiLocation::grandparent()
++                                // Start the interior with the GLobalConsensus junction.
++                                .pushed_front_with_interior(this_global)
++                                .map_err(|_| ProcessMessageError::Unsupported)?
++                                // Finish the interior with the remainder.
++                                .appended_with(actual_origin)
++                                .map_err(|_| ProcessMessageError::Unsupported)?;
++                        }
+					},
+					DescendOrigin(j) => {
+						let Ok(_) = actual_origin.append_with(*j) else {
+							return Err(ProcessMessageError::Unsupported)
+						};
+					},
+					_ => return Ok(ControlFlow::Break(())),
+				};
+				skipped.set(skipped.get() + 1);
+				Ok(ControlFlow::Continue(()))
+			},
+		)?;
+		InnerBarrier::should_execute(
+			&actual_origin,
+			&mut instructions[skipped.get()..],
+			max_weight,
+			properties,
+		)
+	}
+}
+```
+
+#### DescribeFamily
+
+The `DescribeFamily` location descriptor is part of the `HashedDescription` MultiLocation hashing system and exists to describe locations in an easy format for encoding and hashing, so that an AccountId can be derived from this MultiLocation.
+
+The change that's needed in the `DescribeFamily` type is the inclusion of a match arm for absolute locations with the structure `{ parents: 2, interior: [GlobalConsensus(...), Parachain(...), ...] }`.
```

Comment:
Ditto, an absolute location should contain no parents.

Reply:
By a location with no parents, do you mean parents: 0?

Reply:
Yes, and even further than that, I specifically mean it should even only be comprised of `InteriorLocation`s and nothing else.